src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -807,7 +807,7 @@ bool FullyConnected::canBeExecutedInConv1x1() const {
         K = srcDims[inRank - 1];
         N = weightDims[0];
 
-        if (!(M >= 49 && M <= 3136 &&
+        if (!(M >= 2 && M <= 3136 &&
```

Comment:
There is one important aspect that I have missed before. In the case of inner product semantics, all the input tensor dimensions, except the last one, are flattened, so as we have 2D tensor (matrix). That means that actual M dimension is not `srcDims[inRank -2]` only, but a multiplication for all the dimensions from i = 0 to [inRank - 2]. `M = srcDims[0] * srcDims[1] * ... * srcDims[inRank - 2]`. So technically, this is M dimension only in case of 2D input tensor. However, since this heuristic helps us avoid regressions let's leave it as it is, but m.b. it does make sense to rename the variable, so not to confuse other people?

Reply:
Done. Rename M to widthInBrg which should be less confused with inner product semantics.